### PR TITLE
Docs: Add minimal hugo build, update docs README

### PR DESCRIPTION
**Pull request by robbymilo**
**What this PR does / why we need it**:

Adds a stripped-down version of the Grafana website hugo build for docs contributors to use when authoring documentation.

Also updates the docs README with instructions on how to edit the docs.

**Special notes for your reviewer**:

Follow the instructions in `docs/README.md` to get a preview of the website running locally.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,90 +1,41 @@
-### Note for external contributors
+# Building the docs locally
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
# Building the docs locally

When you contribute to documentation, it is a good practice to build the docs on your local machine to make sure your changes appear as you expect. This README explains the process for doing that.
```

File: docs/README.md [hunk 2]
```diff
@@ -1,90 +1,41 @@
...
 
-We are currently working on migrating the docs to a new static website. The `grafana/website` repository will be private during this migration, which unfortunately means the docs site can't be built without access.
+## Requirements
 
-The Markdown content however, is still public in this repository. We still encourage pull requests to make the docs better, and we will make sure the changed content works well on the current docs site. Include any images in your pull request, and we will move them to the `grafana/website` repository.
+Docker >= 2.1.0.3
 
-# Building the docs
+## Getting Started
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
## Build the doc site
```

File: docs/README.md [hunk 3]
```diff
@@ -1,90 +1,41 @@
...
 
-To build the docs locally, you need to have Docker installed. The docs are built using [Hugo](http://gohugo.io/) - a static site generator.
+To build the docs locally:
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion

```

File: docs/README.md [hunk 4]
```diff
@@ -1,90 +1,41 @@
...
 
-**Prepare the Docker image**:
```
Comment (oddlittlebird) [inline docs/README.md]: I suggest using lazy numbering.
```
1.
1.
1.
```

File: docs/README.md [hunk 5]
```diff
@@ -1,90 +1,41 @@
...
+1. Make sure you are in the docs folder: `cd docs`.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
1. In the command line, make sure you are in the docs folder: `cd docs`.
```

File: docs/README.md [hunk 6]
```diff
@@ -1,90 +1,41 @@
...
+2. Run `make build`. This builds the docker image with the correct version of Hugo.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
2. Build the Docker image:

make build

```

File: docs/README.md [hunk 7]
```diff
@@ -1,90 +1,41 @@
...
+3. Run `make run`. This runs the container. It will start a hugo sever, and you can naviagate to the site in your browser at `localhost:1313`.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
1. Run the container:

make run

This command starts a Hugo server.

1. Open your web browser and navigate to http://localhost:1313. 
```

File: docs/README.md (hunk 6 above)
Comment (oddlittlebird) [reply docs/README.md]: Add code tags in the blank lines.

File: docs/README.md (hunk 7 above)
Comment (oddlittlebird) [reply docs/README.md]: Add code tags in the blank lines around `make run`

File: docs/README.md [hunk 8]
```diff
@@ -1,90 +1,41 @@
...
-To build the docs locally, you need to have Docker installed. The docs are built using [Hugo](http://gohugo.io/) - a static site generator.
+To build the docs locally:
 
-**Prepare the Docker image**:
+1. Make sure you are in the docs folder: `cd docs`.
+2. Run `make build`. This builds the docker image with the correct version of Hugo.
+3. Run `make run`. This runs the container. It will start a hugo sever, and you can naviagate to the site in your browser at `localhost:1313`. 
 
-> Due to migration to new static site, the Docker image needs to be built from `old-docs` branch.
+You can then edit files in the `sources` directory which will update `localhost:1313` when changes are saved.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
You can then edit files in the `sources` directory, which updates the content in `localhost:1313` when you save your changes.
```

File: docs/README.md [hunk 9]
```diff
@@ -1,90 +1,41 @@
...
 
-**Build the Documentation**:
+Edit content in the `sources` directory.
 
-Now that the Docker image has been prepared we can build the
-grafana docs and start a docs server.
+### Using `relref` for internal links
 
-If you have not cloned the Grafana repository already then:
+Use the hugo shortcode [relref](https://gohugo.io/content-management/cross-references/#use-ref-and-relref) anytime you are linking to other internal docs pages.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
Use the Hugo shortcode [relref](https://gohugo.io/content-management/cross-references/#use-ref-and-relref) anytime you are linking to other internal docs pages.
```

Comment (oddlittlebird) [reply docs/README.md]: Could you give an example? I found the Hugo documentation very confusing.

File: docs/README.md [hunk 10]
```diff
@@ -1,90 +1,41 @@
...
 
-```
-cd ..
-git clone https://github.com/grafana/grafana
-```
+### Editing the Sidebar Menu
 
-Switch your working directory to the directory this file
-(README.md) is in.
+Edit [sources/menu.yaml](sources/menu.yaml) to make changes to the sidebar. Restart the `make run` command for changes to take effect.
```
Comment (oddlittlebird) [inline docs/README.md]: Please clarify "Restart the `make run` command." Do you mean run it again? Do we need to stop the watch process first?

File: docs/README.md [hunk 11]
```diff
@@ -1,90 +1,41 @@
...
+### Using `relref` for internal links
 
-If you have not cloned the Grafana repository already then:
+Use the hugo shortcode [relref](https://gohugo.io/content-management/cross-references/#use-ref-and-relref) anytime you are linking to other internal docs pages.
 
-```
-cd ..
-git clone https://github.com/grafana/grafana
-```
+### Editing the Sidebar Menu
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
### Edit the side menu
```

File: docs/README.md [hunk 12]
```diff
@@ -1,90 +1,41 @@
...
 
-Switch your working directory to the directory this file
-(README.md) is in.
+Edit [sources/menu.yaml](sources/menu.yaml) to make changes to the sidebar. Restart the `make run` command for changes to take effect.
 
-```
-cd grafana/docs
-```
+### Adding Images
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
### Add images
```

File: docs/README.md [hunk 13]
```diff
@@ -1,90 +1,41 @@
...
 
-An AWS config file is required to build the docs Docker image and to publish the site to AWS. If you are building locally only and do not have any AWS credentials for docs.grafana.org then create an empty file named `awsconfig` in the current directory.
+Images are currently hosted in the grafana/website repo.
```
Comment (oddlittlebird) [inline docs/README.md]: Please expand on this. If this means regular contributors can't update images (right now), then state that plainly.

Also, what should they do if they want to add a new image? Can they update an image by adding a new image to the Grafana repo and changing the link?

File: docs/README.md [hunk 14]
```diff
@@ -1,90 +1,41 @@
...
 
-```
-touch awsconfig
-```
+---
 
-Then run (possibly with `sudo`):
+## Making changes live on grafana.com
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
## Deploy changes to grafana.com
```

File: docs/README.md [hunk 15]
```diff
@@ -1,90 +1,41 @@
...
 
-```
-make watch
-```
+Anytime a PR is merged to master with changes in the `docs` directory, those changes are synced via GitHub action to the grafana/website repo on the `docs-grafana` branch.
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
When a PR is merged to master with changes in the `docs` directory, those changes are automatically synched to the grafana/website repo on the `docs-grafana` branch.
```

File: docs/README.md [hunk 16]
```diff
@@ -1,90 +1,41 @@
...
-### Editing content
-
-Changes to the markdown files should automatically cause a docs rebuild and live reload should reload the page in your browser.
-
-### Troubleshooting
-
-#### Running `make watch` errors out with `Warning: Task "default" not found.`
-
-Ensure that the Docker image from the [website repository](https://github.com/grafana/website) is built using the `old-docs` branch.
+In order to make those changes live, open a PR in the website repo that merges the `docs-grafna` branch into `master`. Then follow the publishing guidelines in that repo.
```
Comment (oddlittlebird) [inline docs/README.md]: I don't think outside contributors can do this. Please clarify this statement. 

Also, is `docs-grafna` correct?

**Review by oddlittlebird — COMMENTED** [review]
Great changes, Robby! I look forward to testing this process out.

I added lots of suggestions to the README doc. Keep in mind that the audience for this is likely to be less technical than the average Grafana user. Technical writers can follow instructions to paste a command into the command line, but you need to explicitly tell them what to do and where. They might never have created a Docker container (I haven't) or know how to run them.

Tech writers often look for open source projects to work on exactly so that they can upgrade their technical skills and gain experience and samples. The easier we can make this process, the more likely they are to do it and contribute again.

File: docs/README.md (hunk 13 above)
Comment (robbymilo) [reply docs/README.md]: This is in the process of changing so I intentionally left it vague - are we moving images back into the repo?

Comment (oddlittlebird) [reply docs/README.md]: Good question! @marcusolsson , do we have an answer yet?

**Review by torkelo — COMMENTED** [review]
This seems a bit messy to maintain, another approach is to build a docker container inside the website repo (like we did before with make docs-build), that contains all the layout & build scripts. When we build that container we can remove all other content and stuff that is not needed. 

That way we do not have to maintain and sync multiple copies of this hugo site/layouts. 

Forgot to add that we would have to publish the docs-base container on docker hub.

**Comment by robbymilo** [comment]
@torkelo My goal here is for external contributors to be able to use this, and they don't have access to the website repo.

**Comment by torkelo** [comment]
@robbymilo I know which is why I said we publish the docker image to docker hub. Then have a docker file that extends it and adds the repo local docs. Like this:
https://github.com/grafana/grafana/blob/master/docs/Dockerfile 

That way we do not need to make website repo public and it's easy to test and run docs site from this repo. And we do not need to maintain multiple hugo sites.

**Comment by torkelo** [comment]
So when we do website doc layout & design changes we only need to rebuild the docs-base image and publish to docker hub

**Comment by marefr** [comment]
I agree with @torkelo - this would be the best setup, especially for external contributors.

File: docs/README.md (hunk 13 above)
Comment (oddlittlebird) [reply docs/README.md]: @daniellee , relevant to our conversation today.

File: docs/README.md [hunk 17]
```diff
@@ -1,90 +1,38 @@
...
+1. In the command line, make sure you are in the docs folder: `cd docs`.
+2. Run `make docs`. This launches a preview of the docs website at `http://localhost:3002/docs/grafana/latest/` which will refresh automatically when changes to content in the `sources` directory are made.
 
-> Due to migration to new static site, the Docker image needs to be built from `old-docs` branch.
+---
 
-Git clone `grafana/website` repo. Run these commands in the root of that repo. **Note** that you may require `sudo`
-when running `make docs-build` depending on how your system's Docker
-service is configured):
+## Content Guidelines
```
Comment (oddlittlebird) [inline docs/README.md]:
```suggestion
## Content guidelines
```

**Review by oddlittlebird — DISMISSED** [review]

**Review by marefr — COMMENTED** [review]
@robbymilo Have tested the new setup using docker image from docker hub and it works as expected :+1: 

One thing to consider, possibly a future improvement, would be to fix the link in header to docs as a convenience.

File: docs/README.md (hunk 13 above)
Comment (torkelo) [reply docs/README.md]: think after this is merged we can work on having images in a separate public repo maybe. We can't have them in this repo due to size

**Review by torkelo — APPROVED** [review]
Great job @robbymilo !

**Pull request merged**